### PR TITLE
SDK-1233. Changes for getBackupName() and setBackupName()

### PR DESCRIPTION
**Pull request by ms-mega**

### CODE REVIEW

File: src/commands.cpp [hunk 1]
```diff
@@ -3730,11 +3732,15 @@ bool CommandGetUserData::procresult(Result r)
         case MAKENAMEID4('*', '!', 'd', 'n'):
             parseUserAttribute(deviceNames, versionDeviceNames);
             break;
-
+
```
Comment (polmr) [inline src/commands.cpp:3735]:
```suggestion

```

File: src/commands.cpp [hunk 2]
```diff
@@ -8201,7 +8223,45 @@ bool CommandBackupPut::procresult(Result r)
...
+        {
+            // serialize and encrypt the TLV container
+            std::unique_ptr<std::string> container(tlv->tlvRecordsToContainer(client->rng, &client->key));
+            client->putua(attrtype, (byte *)container->data(), unsigned(container->size()));
+        }
+        else
+        {
+            LOG_err << "Failed to merge with existing backup names with the new one for backup id: " << backupId;
+        }
+    }
```
Comment (polmr) [inline src/commands.cpp:8264]: this shouldn't be happening in case backupId is UNDEF, shouldn't it?

File: src/heartbeats.cpp [hunk 3]
```diff
@@ -541,7 +552,7 @@ void MegaBackupMonitor::registerBackupInfo(const MegaBackupInfo &info)
     string localFolderEncrypted(mClient->cypherTLVTextWithMasterKey("lf", info.localFolder()) );
     string deviceIdHash = mClient->getDeviceidHash();
 
-    mClient->reqs.add(new CommandBackupPut(mClient, info.type(), info.megaHandle(),
+    mClient->reqs.add(new CommandBackupPut(mClient, info.type(), info.backupName(),info.megaHandle(),
```
Comment (polmr) [inline src/heartbeats.cpp:555]:
```suggestion
    mClient->reqs.add(new CommandBackupPut(mClient, info.type(), info.backupName(), info.megaHandle(),
```

File: src/heartbeats.cpp [hunk 4]
```diff
@@ -430,6 +436,11 @@ BackupType MegaBackupInfoSync::getSyncType(MegaClient *client, const MegaSync &s
     return BackupType::INVALID;
 }
 
+string MegaBackupInfoSync::getSyncName(MegaClient* client, const MegaSync& sync)
```
Comment (polmr) [inline src/heartbeats.cpp:439]: unused?

Comment (sergiohs84) [reply src/heartbeats.cpp]: Removed :+1:

File: src/heartbeats.cpp (hunk 3 above)
Comment (sergiohs84) [reply src/heartbeats.cpp]: Done

File: src/commands.cpp (hunk 2 above)
Comment (sergiohs84) [reply src/commands.cpp]: Definitely. Good catch. Fixed

File: tests/integration/SdkTest_test.cpp [hunk 5]
```diff
@@ -4853,27 +4847,34 @@ TEST_F(SdkTest, SdkBackupNames)
...
+    ASSERT_EQ(attributeValue, backupName.get()) << "getBackupName returned incorrect value";
 
     // give a new name to the backup
-    //err = synchronousSetBackupName(0, mBackupId, "BackupNamesTest");
-    //ASSERT_EQ(MegaError::API_OK, err) << "setBackupName failed (error: " << err << ")";
+    mApi[0].userUpdated = false;
+    backup = "/SdkBackupNames_setBackupName_Test";
+    err = synchronousSetBackupName(0, mBackupId, backup);
+    ASSERT_EQ(MegaError::API_OK, err) << "setBackupName failed (error: " << err << ")";
+    ASSERT_TRUE(waitForResponse(&mApi[0].userUpdated));
```
Comment (sergiohs84) [inline tests/integration/SdkTest_test.cpp:4860]: You should not need to wait for response in this case. Once the `setBackupName()` finishes, the attribute is available (in contrast to the silent automatic setup on the background for the `setBackup()` call.

**Review by sergiohs84 — CHANGES_REQUESTED** [review]

**Comment by sergiohs84** [comment]
trigger compilation

**Comment by ms-mega** [comment]
trigger compilation

**Comment by sergiohs84** [comment]
> megasdk_PR FAILED (3373) -1

It failed due to unrelated error on CloudRaid download:
```
[ RUN      ] SdkTest.SdkTestCloudraidTransferWithConnectionFailures
tests/integration/SdkTest_test.cpp:4123: Failure
Value of: waitForResponse(&mApi[0].transferFlags[MegaTransfer::TYPE_DOWNLOAD], 180)
  Actual: false
Expected: true
Cloudraid download with 404 and 403 errors time out (180 seconds)
[  FAILED  ] SdkTest.SdkTestCloudraidTransferWithConnectionFailures (195303 ms)
```

**Review by sergiohs84 — APPROVED** [review]

**Pull request merged**